### ISSUE TITLE
xiSCSI modules available but need community assistance

### ISSUE THREAD

**Issue by mgreenegit** [PowerShell/DscResources#216]
Two new modules for Windows iSCSI are available but need assistance from the community. Review, tests, and docs.
[https://github.com/robwi/xiSCSIServer](https://github.com/robwi/xiSCSIServer)
[https://github.com/robwi/xiSCSIInitiator](https://github.com/robwi/xiSCSIInitiator)

HELP WANTED

**Comment by PlagueHO** [comment]
Hi @mgreenegit - iSCSI functionality is covered in [this HQRM module](https://github.com/PlagueHO/iSCSIDsc) that is currently going through the [review process](https://github.com/PlagueHO/iSCSIDsc/pull/16) for inclusion in the resource kit. It should be in the final stages now, but I know the DSC team is swamped so might still be a little while for this to be complete.

I did dig through the resources you linked to and there are a number of features that would be great to be copied across (specifically the MSFT_xWaitForiSCSIVirtualDiskTargetMapping). They will require a non-trivial amount of work to get them to current standards though as you say (tests, docs, code standards etc).

So perhaps a good plan would be to get iSCSIDsc in place and then start a process of merging any missing features/resources from xISCSIServer. But of course, that is just my opinion :grin:

**Comment by mgreenegit** [comment]
I agree with your assessment.  That's the right approach.  Definitely not insignificant.

**Comment by PlagueHO** [comment]
@mgreenegit - sweet. We'll I'm definitely up for helping get these feature sets merged.

**Comment by PlagueHO** [comment]
Hi @mgreenegit - I've now released my version of HQRM iSCSI DSC Resources to PSGallery. Do you still want to keep this one open or could it be closed?

**Comment by johlju** [comment]
@PlagueHO Can you confirm that all resources in both the resource modules above in the issue description is obsolete with the new resources in the resource module https://github.com/PlagueHO/iSCSIDsc?

**Comment by PlagueHO** [comment]
Hi @johlju - I'll have to do a bit of exploring in the resources above. There are definitely some differences.

We've taken a different approach to the structure of the resources: how we create the target's and how Virtual Disks are connected and attached to a Server Target. I allow specifying the Virtual disks to attach to the Server Target in the Server Target resource rather than using a separate mapping resource. I also don't have a resource for configuring the Target Portal (I didn't think it was that useful and using it actually resulted in some limitations I didn't like so not prioritized) but the @RobWi one does, but doesn't specifically configure/control Server Targets - instead relying on the Target Portal to sort it out.

So there is not a 1-1 mapping and not an easy way for comparing features. Also, because there are no examples in @RobWi's one it's a bit hard to figure out the intended way it should be used - so I might be making some assumptions about it's features and behavior.

I could always just copy all @RobWi's in to my resource module (minus Virtual Disk as that seems to be one-one mapping). But @RobWi's will require a lot of work (unit tests, integration tests, examples, code style etc). So it's something I'd have to add to my backlog (it's getting pretty long :grin: ).
